core/cap-0067.md

```diff
@@ -0,0 +1,304 @@
++    SC_ADDRESS_TYPE_LIQUIDITY_POOL = 3
+ };
+ 
+ union SCAddress switch (SCAddressType type)
+@@ -188,6 +190,10 @@ case SC_ADDRESS_TYPE_ACCOUNT:
+     AccountID accountId;
+ case SC_ADDRESS_TYPE_CONTRACT:
+     Hash contractId;
++case SC_ADDRESS_TYPE_CLAIMABLE_BALANCE:
++    ClaimableBalanceID claimableBalanceID;
++case SC_ADDRESS_TYPE_LIQUIDITY_POOL:
++    PoolID liquidityPoolID;
+ };
+ 
+ %struct SCVal;
+```
+
+## Semantics
+
+### Remove the admin from the SAC `mint` and `clawback` events
+
+The `mint` event will look like:
+```
+contract: asset, topics: ["mint", to:Address, sep0011_asset:String], data: amount:i128
+```
+
+The `clawback` event will look like:
+```
+contract: asset, topics: ["clawback", from:Address, sep0011_asset:String], data: amount:i128
+```
+
+### Emit the semantically correct event for a Stellar Asset Contract `transfer` when the issuer is involved 
+
+At the moment, if the issuer is the sender in a Stellar Asset Contract `transfer`, the asset will be minted. If the issuer is the recipient, the asset will be burned. The event emitted in both scenarios, however, is the `transfer` event. This CAP changes that behavior to instead emit the `mint`/`burn` event.
+
+### New Events
+This section will go over the semantics of how the additional `transfer` events are emitted for each operation, as well as the `fee` event emitted for the fee paid by the source account. These events will be emitted through `TransactionMeta`, and will not be hashed into the ledger. Note that
+the `contract` field for these events corresponds to the Stellar Asset Contract address for the respective asset. Note that the Stellar Asset Contract instance is not required to be deployed for the asset. The events will be published using the reserved contract address regardless of deployment status.
+
+#### Fees paid by source account
+
+For each transaction whose source account pays fees for the execution of a transaction, emit an event in the following format:
+```
+contract: native asset, topics: ["fee", from:Address], data: [amount:i128]
+```
+Where from is the account paying the fee, either the fee bump fee account or the tx source account.
+
+#### Payment
+Emit one of the following events -
+
+For a payment not involving the issuer, or if both the sender and receiver are the issuer:
+```
+contract: asset, topics: ["transfer", from:Address, to:Address, sep0011_asset:String], data: amount:i128
+```
+
+When sending from an issuer:
+```
+contract: asset, topics: ["mint", to:Address, sep0011_asset:String], data: amount:i128
+```
+
+When sending to an issuer:
+```
+contract: asset, topics: ["burn", from:Address, sep0011_asset:String], data: amount:i128
+```
+
+#### Path Payment Strict Send / Path Payment Strict Receive
+For each movement of the asset created by the path payment, emit one of the following - 
+
+For a movement not involving the issuer:
+```
+contract: asset, topics: ["transfer", from:Address, to:Address, sep0011_asset:String], data: amount:i128
```

Comment:
This needs clarity on who `from` and `to` are, here

Reply:
I'll address this in a follow up